Added documentation (GeometryInstance and VisualInstance)

TwistedTwigleg opened the pull request:
I've filled in the documentation for both GeometryInstance and VisualInstance.

TwistedTwigleg committed:
Added documentation for GeometryInstance and VisualInstance

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
 	</methods>
 	<members>
 		<member name="cast_shadow" type="int" setter="set_cast_shadows_setting" getter="get_cast_shadows_setting" enum="GeometryInstance.ShadowCastingSetting">
+			The currently selected shadow casting flag.
```
Whenever possible, remove extra words that don't add content, ex: "currently".

When a member uses named constants, the standard has been to refer to them in the description, so the reader knows where to look. "See SHADOW_CASTING_SETTING_* constants for values."

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			The currently selected shadow casting flag.
 		</member>
 		<member name="extra_cull_margin" type="float" setter="set_extra_cull_margin" getter="get_extra_cull_margin">
+			The extra distance added to the GeometryInstance's bounding box ([AABB]) to increase it's cull box.
```
it's -> its

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			The extra distance added to the GeometryInstance's bounding box ([AABB]) to increase it's cull box.
 		</member>
 		<member name="lod_max_distance" type="float" setter="set_lod_max_distance" getter="get_lod_max_distance">
+			The max distance for the LOD of this GeometryInstance.
```
Per doc guidelines, prefer possessive over "of the" constructions. 

The GeometryInstance's max LOD distance.

akien-mga commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
 		</member>
 		<member name="material_override" type="Material" setter="set_material_override" getter="get_material_override">
+			The material override for the whole geometry.
+
```
Please don't add empty lines, one line == one paragraph already. Use `git commit --amend` to modify that commit (see [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html)).

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			The max distance for the LOD of this GeometryInstance.
 		</member>
 		<member name="lod_max_hysteresis" type="float" setter="set_lod_max_hysteresis" getter="get_lod_max_hysteresis">
+			The max margin for the LOD of this GeometryInstance.
```
See previous comment.

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			The max margin for the LOD of this GeometryInstance.
 		</member>
 		<member name="lod_min_distance" type="float" setter="set_lod_min_distance" getter="get_lod_min_distance">
+			The min distance for the LOD of this GeometryInstance.
```
See previous comment.

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			The min distance for the LOD of this GeometryInstance.
 		</member>
 		<member name="lod_min_hysteresis" type="float" setter="set_lod_min_hysteresis" getter="get_lod_min_hysteresis">
+			The min margin for the LOD of this GeometryInstance.
```
See previous comment.

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
 		<member name="material_override" type="Material" setter="set_material_override" getter="get_material_override">
+			The material override for the whole geometry.
+
+			If there is a material in material_override, it will be used instead of any material set in any material slot of the mesh.
```
Don't use blank lines. Newlines start new paragraphs.

cbscribe commented on doc/classes/GeometryInstance.xml:
```diff
@@ -14,32 +14,54 @@
+			If there is a material in material_override, it will be used instead of any material set in any material slot of the mesh.
 		</member>
 		<member name="use_in_baked_light" type="bool" setter="set_flag" getter="get_flag">
+			If [code]true[/code], then this GeometryInstance will be used when baking lights using a [GIProbe] and/or any other form of baked lighting.
```
Remove comma after "If [code]true[/code]"
Remove "then"

cbscribe commented on doc/classes/VisualInstance.xml:
```diff
@@ -13,12 +13,14 @@
 			<return type="AABB">
 			</return>
 			<description>
+				Returns the transformed [AABB] (also known as the bounding box) for this VisualInstance.
```
Might be useful to indicate what "transformed" means here.

cbscribe commented on doc/classes/VisualInstance.xml:
```diff
@@ -27,11 +29,17 @@
 			<argument index="0" name="base" type="RID">
 			</argument>
 			<description>
+				Allows you to change the base of te VisualInstance, which changes how the engine handels the VisualInstance under the hood.
```
te -> the
handels -> handles

"Allows you to change" -> Sets

cbscribe commented on doc/classes/VisualInstance.xml:
```diff
@@ -27,11 +29,17 @@
 		<member name="layers" type="int" setter="set_layer_mask" getter="get_layer_mask">
+			The render layer(s) this VisualInstance is drawn on.
+
+			This object will only be visible for [Camera]'s whose cull mask includes the render object this VisualInstance is set to.
```
"[Camera]'s" - remove apostrophe, this isn't possessive.

cbscribe reviewed (changes requested):
Great work! I added some corrections and a couple of requests for clarification.

TwistedTwigleg commented:
Thanks @cbscribe  and @akien-mga!

I fixed the problems in VisualInstance and GeometeryInstance. Now they hopefully have no more spelling and grammar issues, along with being more clear. I fixed them in the second commit, is that okay? I can try to fix the first commit if needed.

I added documentation for World, Theme, StreamTexture, Sky, and ShortCut. They probably could use a second pair of eyes, since I'm not sure everything is 100% accurate to what they actually do (mainly Sky and Shortcut). I tried to assure they don't have problems like VisualInstance and GeometeryInstance, so hopefully everything is good there.

cbscribe commented:
You will need to squash down to one commit before this can be merged - the Godot team likes to keep the commit log clean.

Also, as a general rule of thumb, try and do a separate PR for different, unrelated changes. It's easier to track and review if they're not all in one gigantic commit.

TwistedTwigleg commented:
Should I squash this commit into one? Or make another PR with the additional documentation? (World, Theme, StreamTexture, Sky, and ShortCut)

cbscribe commented:
Thanks for your effort!
